### PR TITLE
Add an Edit Mode button in the Settings.

### PR DESCRIPTION
**Pull request by francoislerall**
Extracted from https://github.com/GreenDelta/olca-app/pull/170.

### CODE REVIEW

File: olca-app/src/org/openlca/app/editors/graphical/GraphConfig.java [hunk 1]
```diff
@@ -13,6 +13,7 @@ public class GraphConfig implements Copyable<GraphConfig> {
 	public boolean showFlowAmounts = true;
 	public boolean showElementaryFlows = false;
 	public boolean isRouted = true;
+	public boolean isProcessEditingEnable = false;
```
[the remark this answers is not in the document]
Comment (msrocka) [reply olca-app/src/org/openlca/app/editors/graphical/GraphConfig.java:16]: better name the field and property `isProcessEditingEnabled`

**Review by msrocka — CHANGES_REQUESTED** [review]
better name the field `isProcessEditingEnabled`; also, the `Add flow` button and actions for modifying processes should be only shown, when this field is set to `true`, currently it does not have any effect, right?

**Comment by francoislerall** [comment]
> better name the field `isProcessEditingEnabled`; also, the `Add flow` button and actions for modifying processes should be only shown, when this field is set to `true`, currently it does not have any effect, right?

Yes, it is just the addition of an edit mode in the config. The implementation deeper in the code is in this https://github.com/GreenDelta/olca-app/pull/173. (it was in the same PR before but in a different commit).

File: olca-app/src/org/openlca/app/editors/graphical/GraphConfig.java [hunk 2]
```diff
@@ -71,6 +74,8 @@ public static GraphConfig fromJson(JsonObject obj) {
 			obj, "showElementaryFlows", false);
 		config.isRouted = Json.getBool(
 				obj, "isRouted", true);
+		config.isProcessEditingEnabled = Json.getBool(
+			obj, "editMode", false);
```
Comment (msrocka) [inline olca-app/src/org/openlca/app/editors/graphical/GraphConfig.java:78]: it should read the property `isProcessEditingEnabled` not `editMode`

**Review by msrocka — CHANGES_REQUESTED** [review]

**Pull request merged**